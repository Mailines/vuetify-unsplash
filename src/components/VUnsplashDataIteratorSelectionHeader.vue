<template>
  <v-container fluid>
    <v-card v-if="selectMultiple && itemsSelected" class="pa-2">
      <v-layout
        row
        class="d-flex align-center px-3 px-md-6 pb-1 justify-center"
      >
        <v-flex class="justify-start shrink">
          <span
            >{{ label.selected }}
            <span class="font-weight-bold">{{ numberOfSelectedItems }}</span>
            {{ label.of }}
            <span class="font-weight-bold">{{ maxMultipleSelect }}</span>
          </span>
        </v-flex>
        <v-divider vertical class="mx-4"></v-divider>
        <v-flex class="justify-start">
          <v-progress-circular
            :value="(numberOfSelectedItems / maxMultipleSelect) * 100"
            :color="accentColor"
          ></v-progress-circular>
        </v-flex>
        <v-spacer></v-spacer>
        <v-flex class="d-flex justify-end align-center px-2">
          <v-tooltip bottom>
            <template v-slot:activator="{ on }">
              <v-btn
                icon
                :color="accentColor"
                v-on="on"
                @click="unselectAllImages"
                class="mr-1"
                ><v-icon>mdi-select-off</v-icon></v-btn
              >
            </template>
            <span>{{ label.cancelSelection }}</span>
          </v-tooltip>
          <v-tooltip
            v-if="selectMultiple && maxMultipleSelect >= numberOfImages"
            bottom
          >
            <template v-slot:activator="{ on }">
              <v-btn
                icon
                :color="accentColor"
                v-on="on"
                @click="selectAllImages"
                class="ml-1"
                ><v-icon>mdi-select-all</v-icon></v-btn
              >
            </template>
            <span>{{ label.selectAll }}</span>
          </v-tooltip>
        </v-flex>
      </v-layout>
      <v-divider></v-divider>
      <v-layout row class="justify-start px-6 pt-1">
        <v-flex v-for="(image, i) in images" :key="i" class="shrink">
          <v-chip
            v-if="image.selected"
            close
            pill
            class="ma-1"
            @click:close="unselectImageById(image.id)"
          >
            <v-avatar left>
              <v-img :src="image.urls.thumb"></v-img>
            </v-avatar>
          </v-chip>
        </v-flex>
      </v-layout>
    </v-card>
  </v-container>
</template>

<script>
export default {
  props: {
    itemsSelected: {
      type: Boolean,
      required: true
    },
    selectMultiple: {
      type: Boolean,
      required: true
    },
    label: {
      type: Object,
      required: true
    },
    maxMultipleSelect: {
      type: Number,
      required: true
    },
    numberOfSelectedItems: {
      type: Number,
      required: true
    },
    accentColor: {
      type: String,
      required: true
    },
    numberOfImages: {
      type: Number,
      required: true
    },
    images: {
      type: Array,
      required: true
    }
  },
  methods: {
    unselectAllImages() {
      this.$emit("unselectAllImages");
    },
    unselectImageById(imageId) {
      this.$emit("unselectImageById", imageId);
    },
    selectAllImages() {
      this.$emit("selectAllImages");
    }
  }
};
</script>
