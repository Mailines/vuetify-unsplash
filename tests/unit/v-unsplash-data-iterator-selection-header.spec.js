import VUnsplashDataIteratorSelectionHeader from "../../src/components/VUnsplashDataIteratorSelectionHeader";
import { mount, createLocalVue } from "@vue/test-utils";
import Vuetify from "vuetify";

describe("VUnsplashDataIteratorSelectionHeader.vue", () => {
  let localVue;
  let wrapper;
  let vuetify;

  beforeEach(() => {
    localVue = createLocalVue();
    vuetify = new Vuetify();
    wrapper = mount(VUnsplashDataIteratorSelectionHeader, {
      localVue,
      vuetify,
      propsData: {
        accentColor: "#1976d2",
        itemsSelected: true,
        selectMultiple: true,
        label: {
          of: "of",
          selectAll: "select all",
          cancelSelection: "cancel selection",
          selected: "selected"
        },
        maxMultipleSelect: 2,
        numberOfSelectedItems: 1,
        numberOfImages: 10,
        images: [
          {
            id: "123456",
            selected: true,
            urls: {
              thumb:
                "https://www.google.com/images/branding/googlelogo/2x/googlelogo_color_272x92dp.png"
            }
          }
        ]
      }
    });
  });
  it("renders and match snapshot", async () => {
    expect(wrapper.exists()).toBe(true);
    expect(wrapper.html()).toMatchSnapshot();
  });
  it("emits unselectAllImages event when calling unselectAllImages()", async () => {
    wrapper.vm.unselectAllImages();
    expect(wrapper.emitted().unselectAllImages).toBeTruthy();
  });
  it("emits selectAllImages event when calling selectAllImages()", async () => {
    wrapper.vm.selectAllImages();
    expect(wrapper.emitted().selectAllImages).toBeTruthy();
  });
  it("emits unselectImageById  event with id of image when calling unselectImageById(imageId)", async () => {
    const imageId = 123456768;
    wrapper.vm.unselectImageById(imageId);
    expect(wrapper.emitted().unselectImageById[0][0]).toBe(imageId);
  });
});
