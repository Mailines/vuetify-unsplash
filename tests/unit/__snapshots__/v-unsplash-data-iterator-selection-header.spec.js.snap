// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VUnsplashDataIteratorSelectionHeader.vue renders and match snapshot 1`] = `
<div class="container container--fluid">
  <div class="pa-2 v-card v-sheet theme--light">
    <div class="layout d-flex align-center px-3 px-md-6 pb-1 justify-center row">
      <div class="flex justify-start shrink"><span>selected
          <span class="font-weight-bold">1</span>
        of
        <span class="font-weight-bold">2</span></span></div>
      <hr role="separator" aria-orientation="vertical" class="mx-4 v-divider v-divider--vertical theme--light">
      <div class="flex justify-start">
        <div role="progressbar" aria-valuemin="0" aria-valuemax="100" aria-valuenow="50" class="v-progress-circular" style="height: 32px; width: 32px; color: rgb(25, 118, 210); caret-color: #1976d2;"><svg xmlns="http://www.w3.org/2000/svg" viewBox="22.857142857142858 22.857142857142858 45.714285714285715 45.714285714285715" style="transform: rotate(0deg);">
            <circle fill="transparent" cx="45.714285714285715" cy="45.714285714285715" r="20" stroke-width="5.714285714285714" stroke-dasharray="125.664" stroke-dashoffset="0" class="v-progress-circular__underlay"></circle>
            <circle fill="transparent" cx="45.714285714285715" cy="45.714285714285715" r="20" stroke-width="5.714285714285714" stroke-dasharray="125.664" stroke-dashoffset="62.83185307179586px" class="v-progress-circular__overlay"></circle>
          </svg>
          <div class="v-progress-circular__info"></div>
        </div>
      </div>
      <div class="spacer"></div>
      <div class="flex d-flex justify-end align-center px-2"><button type="button" class="mr-1 v-btn v-btn--flat v-btn--icon v-btn--round theme--light v-size--default" style="color: rgb(25, 118, 210); caret-color: #1976d2;"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate mdi mdi-select-off theme--light"></i></span></button><span class="v-tooltip v-tooltip--bottom"><!----></span>
        <!---->
      </div>
    </div>
    <hr role="separator" aria-orientation="horizontal" class="v-divider theme--light">
    <div class="layout justify-start px-6 pt-1 row">
      <div class="flex shrink"><span class="ma-1 v-chip v-chip--no-color v-chip--pill v-chip--removable theme--light v-size--default"><span class="v-chip__content"><div class="v-avatar v-avatar--left" style="height: 48px; min-width: 48px; width: 48px;"><div class="v-image v-responsive theme--light"><transition-stub name="fade-transition" mode="in-out"><div class="v-image__image v-image__image--preload v-image__image--cover" style="background-position: center center;"></div></transition-stub><div class="v-responsive__content"></div></div></div><button type="button" class="v-icon notranslate v-chip__close v-icon--link v-icon--right mdi mdi-close-circle theme--light" style="font-size: 18px;"></button></span></span></div>
    </div>
  </div>
</div>
`;
